examples/integration_test.py

```diff
@@ -565,6 +565,175 @@ def my_on_revoke(consumer, partitions):
+    # Since we're not making use of statistics the low offset is not known so ignore it.
+    lo, hi = c.get_watermark_offsets(assignment[0], cached=True)
+    print('Cached offsets for %s: %d - %d' % (assignment[0], lo, hi))
+
+    # Query broker for offsets
+    lo, hi = c.get_watermark_offsets(assignment[0], timeout=1.0)
+    print('Queried offsets for %s: %d - %d' % (assignment[0], lo, hi))
+
+    # Close consumer
+    c.close()
+
+    # Start a new client and get the committed offsets
+    c = confluent_kafka.Consumer(**conf)
+    offsets = c.committed(list(map(lambda p: confluent_kafka.TopicPartition(topic, p), range(0, 3))))
+    for tp in offsets:
+        print(tp)
+
+    c.close()
+
+
+def verify_batch_consumer_performance():
+    """ Verify batch Consumer performance """
+
+    conf = {'bootstrap.servers': bootstrap_servers,
+            'group.id': uuid.uuid1(),
+            'session.timeout.ms': 6000,
+            'error_cb': error_cb,
+            'default.topic.config': {
+                'auto.offset.reset': 'earliest'
+            }}
+
+    c = confluent_kafka.Consumer(**conf)
+
+    def my_on_assign(consumer, partitions):
+        print('on_assign:', len(partitions), 'partitions:')
+        for p in partitions:
+            print(' %s [%d] @ %d' % (p.topic, p.partition, p.offset))
+        consumer.assign(partitions)
+
+    def my_on_revoke(consumer, partitions):
+        print('on_revoke:', len(partitions), 'partitions:')
+        for p in partitions:
+            print(' %s [%d] @ %d' % (p.topic, p.partition, p.offset))
+        consumer.unassign()
+
+    c.subscribe([topic], on_assign=my_on_assign, on_revoke=my_on_revoke)
+
+    max_msgcnt = 1000000
+    bytecnt = 0
+    msgcnt = 0
+    batch_size = 1000
+
+    print('Will now consume %d messages' % max_msgcnt)
+
+    if with_progress:
+        bar = Bar('Consuming', max=max_msgcnt,
+                  suffix='%(index)d/%(max)d [%(eta_td)s]')
+    else:
+        bar = None
+
+    while True:
+        # Consume until EOF or error
+
+        msglist = c.consume(num_messages=batch_size, timeout=20.0)
+
+        for msg in msglist:
+            if msg.error():
+                if msg.error().code() == confluent_kafka.KafkaError._PARTITION_EOF:
+                    # Reached EOF for a partition, ignore.
+                    continue
+                else:
+                    raise confluent_kafka.KafkaException(msg.error())
+
+            bytecnt += len(msg)
+            msgcnt += 1
+
+            if bar is not None and (msgcnt % 10000) == 0:
+                bar.next(n=10000)
+
+            if msgcnt == 1:
+                t_first_msg = time.time()
+            if msgcnt >= max_msgcnt:
+                break
+
+        if msgcnt >= max_msgcnt:
+            break
+
+    if bar is not None:
+        bar.finish()
+
+    if msgcnt > 0:
+        t_spent = time.time() - t_first_msg
+        print('%d messages (%.2fMb) consumed in %.3fs: %d msgs/s, %.2f Mb/s' %
```

Comment:
what thruput are you getting with the batch interface, compared to the standard one?

Reply:
Having trouble running the integration test, but running a quick test I wrote on an actual Kafka stream I got:
```
consume took 1.33 seconds
1000000 messages, 662.56 MB, 497.54 MB/sec

poll tool 2.17 seconds
1000000 messages, 662.62 MB, 304.57 MB/sec
```
So definitely an improvement, though I found I had to tune the batch size to get the best results, e.g. 1k messages performs better than 10k messages or 100 messages.

Reply:
damn!